### PR TITLE
update XLA nightly version check

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

it seems XLA will have 1.8 soon and they already renamed nightly to 1.9

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: dockers/base-xla/Dockerfile [hunk 1]
```diff
@@ -115,4 +115,4 @@ RUN \
     conda info && \
     pip list && \
     python -c "import sys; assert sys.version[:3] == '$PYTHON_VERSION', sys.version" && \
-    python -c "import torch; ver = '$XLA_VERSION' ; ver = dict(nightly='1.8').get(ver, ver) ; assert torch.__version__[:3] == ver, torch.__version__"
+    python -c "import torch; ver = '$XLA_VERSION' ; ver = dict(nightly='1.9').get(ver, ver) ; assert torch.__version__[:3] == ver, torch.__version__"
```
Comment (carmocca) [inline dockers/base-xla/Dockerfile:118]: I don't understand what this does...

`dict(nightly='1.9').get(ver, ver)`

How does this ever return anything different than `"$XLA_VERSION"`?

Comment (Borda) [reply dockers/base-xla/Dockerfile:118]: this is our consistency check that you have really installed what you requested
the mapping trick is because you download it nightly without an explicit version number but after installation, it has a specific number, in this case when you download nightly you get 1.9

Comment (carmocca) [reply dockers/base-xla/Dockerfile:118]: got it.

maybe this would be simpler

```suggestion
    python -c "import torch; ver = '$XLA_VERSION' ; ver = '1.9' if ver == 'nightly' else ver ; assert torch.__version__[:3] == ver, torch.__version__"
```

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
